### PR TITLE
Add method to return signed URL and headers

### PR DESCRIPTION
**Pull request by waynelee**
Adds an API that returns just the URL and signed headers without sending the request to the server.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -52,6 +52,36 @@ OSS.prototype.generateSign = function(req, resource) {
...
+    method: options.method,
+    getHeader: function(key) { return headers[key]; },
+    _headers: headers
+  };
+
+  // Sign headers
+  headers['Authorization'] = this.generateSign(req, getResource(options));
+
+  // Generate URL
+  var url = require('url').format({
```
Comment (coderhaoxin) [inline index.js]: put `require('url')` to the top?

File: index.js [hunk 2]
```diff
@@ -52,6 +52,36 @@ OSS.prototype.generateSign = function(req, resource) {
 };
 
 /**
+ * @param {object} - options
+ *   required properties: method, bucket, object, headers
+ */
+OSS.prototype.getSignedUrl = function(options) {
+  var headers = {};
+  Object.assign(headers, options.headers);
```
Comment (coderhaoxin) [inline index.js]:
```js
var headers = Object.assign({}, options.headers);
```

**Comment by coderhaoxin** [comment]
@waynelee add a test case for this?

**Comment by coderhaoxin** [comment]
@waynelee Do you or someone else want to maintain this repo? I'll add a collaborator

**Comment by waynelee** [comment]
@coderhaoxin. Updated new commit that addresses your comments. Also included a few extra changes to make the test work in my environment. We won't be doing much with Aliyun, it's just some exploratory work right now, so I don't think I'll be a good candidate to be a maintainer. I'd be happy to be a contributor if you'd like. Thanks for this module, it was very helpful!

**Comment by coderhaoxin** [comment]
landed 92679471ed4fc33ccd24c6fa3cd1a38722130bd2, published as `v1.0.0`

**Comment by coderhaoxin** [comment]
@waynelee collaborator added.

**Pull request closed without merging**